src/components/Carousel/index.tsx

```diff
@@ -1,88 +1,53 @@
-import ReactIdSwiper from '@class101/swiper';
-import { ReactIdSwiperProps, SwiperInstance } from '@class101/swiper/lib/types';
 import React, { PureComponent } from 'react';
-import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
-import { AutoplayOptions, PaginationOptions, SwiperEvent } from 'swiper';
 
-import { media, SIZES } from '../../core/BreakPoints';
-import { gray700, white } from '../../core/Colors';
-import Navigation, { CarouselNavigationPosition } from './Navigation';
+import { SIZES } from '../../core/BreakPoints';
+import { Swiper, SwiperInstance, SwiperNavigationPosition, SwiperPaginationTheme, SwiperProps } from '../Swiper';
 import ViewAllButton from './ViewAllButton';
 
 type SlidesPerView = 'auto' | 1 | 2 | 3 | 4 | 6;
 
-export enum CarouselPaginationTheme {
-  Dark,
-  Light,
-}
-
-export interface CarouselProps {
-  navigationContainerMaxWidth?: number;
-  navigation: boolean;
-  navigationPosition: CarouselNavigationPosition;
-  pagination: boolean;
-  paginationTheme: CarouselPaginationTheme;
+type UsingSwiperProps = Pick<
+  SwiperProps,
+  | 'containerContentMaxWidth'
+  | 'hasPagination'
+  | 'hasNavigation'
+  | 'paginationTheme'
+  | 'freeMode'
+  | 'autoplay'
+  | 'loop'
+  | 'customNavigation'
+  | 'customPagination'
+  | 'children'
+  | 'slideToClickedSlide'
+  | 'speed'
+  | 'getSwiperInstance'
+  | 'observer'
+  | 'lgSlidesSideOffset'
+  | 'smSlidesSideOffset'
+>;
+
+export interface OwnCarouselProps {
   lgSlidesPerView: SlidesPerView;
   smSlidesPerView: SlidesPerView;
-  lgSlidesSideOffset: number;
-  smSlidesSideOffset: number;
-  freeMode: boolean;
   lgSpaceBetween?: boolean;
   smSpaceBetween?: boolean;
-  autoplay?: AutoplayOptions | boolean;
-  on?: { [key in SwiperEvent]?: () => void };
-  loop: boolean;
-  shouldSwiperUpdate?: boolean;
-  rebuildOnUpdate?: boolean;
-  observer?: boolean;
-  observeParents?: boolean;
   className?: string;
-  slideToClickedSlide?: boolean;
-  children: React.ReactNode;
   onChangeSlides?: (index: number) => void;
   activeSlideIndex?: number;
 }
 
-const DEFAULT_PARAMS = {
-  threshold: 10,
-};
-
-const DEFAULT_PAGINATION_PARAMS: { [key in string]: PaginationOptions } = {
-  pagination: {
-    el: '.swiper-pagination',
-    type: 'bullets',
-    clickable: true,
-  },
-};
+export type CarouselProps = UsingSwiperProps & OwnCarouselProps;
```

Comment:
```suggestion
export type CarouselProps = {
  swiperProps: UsingSwiperProps;
  ...나머지 OwnCarouselProps
}
```

이렇게 하는 것이 Swiper prop을 나눠서 관리할 수 있어서 가독성이 좋고, 미래에 swiper의 prop 변경 때문에 own carousel prop과 이름이 겹치는 일이 없습니다

이 변경을 적용하면 `getSwiperParams()` 을 쉽게 제거할 수 있습니다